### PR TITLE
[THREESCALE-7475] Api docs calls won't destroy user session because of CSRF

### PR DESCRIPTION
**Pull request by jsmadis**
**What this PR does / why we need it**:

Adds CSRF token to the HTTP calls that are made through the 3scale API Docs.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-7475

### CODE REVIEW

**Review by akostadinov — DISMISSED** [review]

**Comment by hallelujah** [comment]
Maybe the calls should not return a `Set-Cookie` headers

That is a viable solution too.
I don't think any API request should use cookies/sessions as they have token to identify the user

**Comment by jsmadis** [comment]
> I don't think any API request should use cookies/sessions as they have token to identify the user

@hallelujah, I agree, but that's the point of the other issue. We can postpone this PR, and I will try to fix our API request to ignore the cookies/sessions. Then we can see if it's a viable solution for this problem also.

**Comment by hallelujah** [comment]
What is the other issue? I think (if not mistaken) we should use 
 this for all APIs https://api.rubyonrails.org/classes/ActionController/RequestForgeryProtection/ClassMethods.html#method-i-skip_forgery_protection

I wonder we do not have it though
https://github.com/3scale/porta/blob/b2fd9c02596ce109f17d692475384c4aae7259d2/app/controllers/admin/api/base_controller.rb

**Comment by hallelujah** [comment]
So it is because we use a completely wrong logic

https://github.com/3scale/porta/blob/master/app/controllers/application_controller.rb#L18-L20

**Comment by jsmadis** [comment]
Oh, I see. Thanks @hallelujah 

The fix is even easier then.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ class ApplicationController < ActionController::Base
 
   # Disable CSRF protection for non xml requests.
   skip_before_action :verify_authenticity_token, if: -> do
-    (params.key?(:provider_key) || params.key?(:access_token)) && request.format.xml?
+    (params.key?(:provider_key) || params.key?(:access_token)) && !request.format.xml?
```
Comment (hallelujah) [inline app/controllers/application_controller.rb:19]:
```suggestion
    (params.key?(:provider_key) || params.key?(:access_token)) && is_api_controller?
```

Then create a method 
```
class ApplicationController < ActionController::Base
  def api_controller?
    false
  end
end

# And in other :real" API base controllers set it to true

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -14,9 +14,9 @@ class ApplicationController < ActionController::Base
 
   protect_from_forgery with: :reset_session # See ActionController::RequestForgeryProtection for details
 
-  # Disable CSRF protection for non xml requests.
+  # Disable CSRF protection for requests to REST API.
   skip_before_action :verify_authenticity_token, if: -> do
-    (params.key?(:provider_key) || params.key?(:access_token)) && request.format.xml?
+    (params.key?(:provider_key) || params.key?(:access_token)) && api_controller?
```
Comment (akostadinov) [inline app/controllers/application_controller.rb:19]: probably worthless micro-optimization, but shouldn't we first check `api_controller?` and then the other checks?

**Review by akostadinov — APPROVED** [review]

**Review by hallelujah — APPROVED** [review]

**Pull request merged**